src/components/BudgetExpiryAlertAndModal/data/utils.js

```diff
@@ -5,9 +5,12 @@ import ExpiryThresholds from './expiryThresholds';
 
 dayjs.extend(duration);
 
-export const getNonExpiredBudgets = (budgets) => {
+export const getExpiredAndNonExpiredBudgets = (budgets) => {
   const today = dayjs();
-  return budgets.filter((budget) => today <= dayjs(budget.end));
+  return {
+    nonExpiredBudgets: budgets.filter((budget) => today <= dayjs(budget.end)),
+    expiredBudgets: budgets.filter((budget) => today > dayjs(budget.end)),
```

Comment:
[suggestion] I wonder if it might make sense to have a single iteration through `budgets` rather than running `.filter` twice (iterating through the array twice).

```jsx
const nonExpiredBudgets = [];
const expiredBudgets = [];
budgets.forEach((budget) => {
  if (today <= dayjs(budget.end)) {
    nonExpiredBudgets.push(budget);
  } else {
    expiredBudgets.push(budget);
  }
});
```